### PR TITLE
Dashboards: Migrate from aria-label e2e selectors to data-testid

### LINKED ISSUE #78409: Dashboards: Migrate from aria-label e2e selectors to data-testid
Hi @grafana/dashboards-squad!

A while back to changed to prefer using `data-testid` for E2E selectors instead of `aria-label`. There's only a few of these old `aria-label`-based selectors left, so we've made this list of files that your team owns so we can tidy them up 🙌.

For some guidance on how exactly to fix these, check out [this comment](https://github.com/grafana/grafana/issues/36523#issuecomment-1819160512).

There are 46 issues over 27 files:

- [ ] 1 issue in `public/app/features/dashboard-scene/inspect/InspectJsonTab.tsx` 
- [ ] 4 issues in `public/app/features/dashboard/components/AddPanelWidget/AddPanelWidget.tsx` 
- [ ] 4 issues in `public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx` 
- [ ] 1 issue in `public/app/features/dashboard/components/PanelEditor/OptionsPane.tsx` 
- [ ] 2 issues in `public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx` 
- [ ] 1 issue in `public/app/features/dashboard/components/PanelEditor/OptionsPaneItemDescriptor.tsx` 
- [ ] 3 issues in `public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx` 
- [ ] 1 issue in `public/app/features/dashboard/components/PanelEditor/VisualizationSelectPane.tsx` 
- [ ] 2 issues in `public/app/features/dashboard/components/SaveDashboard/SaveDashboardButton.tsx` 
- [ ] 3 issues in `public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardForm.tsx` 
- [ ] 1 issue in `public/app/features/dashboard/containers/DashboardPage.tsx` 
- [ ] 1 issue in `public/app/features/inspector/InspectDataTab.tsx` 
- [ ] 1 issue in `public/app/features/inspector/InspectJSONTab.tsx` 
- [ ] 1 issue in `public/app/features/inspector/InspectStatsTab.tsx` 
- [ ] 2 issues in `public/app/features/inspector/QueryInspector.tsx` 
- [ ] 1 issue in `public/app/features/panel/components/VizTypePicker/PanelTypeCard.tsx` 
- [ ] 2 issues in `public/app/features/playlist/PlaylistForm.tsx` 
- [ ] 1 issue in `public/app/features/query/components/QueryEditorRow.tsx` 
- [ ] 1 issue in `public/app/features/query/components/QueryEditorRowHeader.tsx` 
- [ ] 2 issues in `public/app/features/query/components/QueryGroup.tsx` 
- [ ] 1 issue in `public/app/features/variables/editor/LegacyVariableQueryEditor.tsx` 
- [ ] 1 issue in `public/app/features/variables/editor/VariableEditorEditor.tsx` 
- [ ] 2 issues in `public/app/features/variables/editor/VariableEditorList.tsx` 
- [ ] 4 issues in `public/app/features/variables/editor/VariableEditorListRow.tsx` 
- [ ] 1 issue in `public/app/features/variables/editor/VariableValuesPreview.tsx` 
- [ ] 1 issue in `public/app/features/variables/pickers/shared/VariableLink.tsx` 
- [ ] 1 issue in `public/app/features/variables/pickers/shared/VariableOptions.tsx`

### PR DESCRIPTION
**Pull request by khushijain21**
This PR fixes #78409

### CODE REVIEW

**Comment by khushijain21** [comment]
This PR fixes a few files. Let me know if this is okay. Will continue for the rest

File: public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx [hunk 1]
```diff
@@ -115,15 +115,15 @@ export const OptionsPaneCategory = React.memo(
         className={boxStyles}
         data-testid="options-category"
         data-plugin-sandbox={sandboxId}
-        aria-label={selectors.components.OptionsGroup.group(id)}
+        data-testid={selectors.components.OptionsGroup.group(id)}
```
Comment (joshhunt) [inline public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx:118]: Duplicate `data-testid` on this div. It's difficult to know which is the least disruptive one to remove...

Lets remove `data-testid="options-category"` from this.

**Review by joshhunt — CHANGES_REQUESTED** [review]
Thanks for the contribution! Changes look good - just one duplicate prop that neeeds sorting out.

---

Because this pull request is so large and it has so many disparate changes to the `.betterer.results` file, it's going to hard to merge this in without merge conflicts. 

There's also no way the unit tests are all passing, but we can't get to that step yet in CI because it's failing at either merge conflicts or linting (due to the duplicate prop).

This is why I wish this was a series of smaller PRs that would have made this a lot more manageable...

File: public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx (hunk 1 above)
Comment (khushijain21) [reply public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx:118]: did this, can you direct me on how you are checking all the tests locally?

Comment (khushijain21) [reply public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx]: about breaking this PR into series of smaller PR's. Will this work for now? or does it require it?

**Comment by joshhunt** [comment]
@khushijain21 It's harder to make big PRs with many seperate changes. It would be easier for you if the pull request was smaller, as it's easier and quicker to run unit tests locally, and you'll get less conflicts, especially on the .betterer.results file.

It's easier for _us_ to review small pull requests, so it'll get merged quicker, which is also better for you :)

You can run unit tests locally with `yarn test`. If your changes are not comitted yet (you run `git status` and you can see the changed files), this will [attempt to test only the files you changed](https://jestjs.io/docs/cli#--onlychanged). Otherwise, you can run unit tests locally with `yarn test <path to what needs testing>`. e.g. `yarn test public/app/features/dashboard` to run tests matching that name https://jestjs.io/docs/cli#running-from-the-command-line

**Pull request closed without merging**